src/github.com/getlantern/flashlight/config/config.go

```diff
@@ -388,26 +388,25 @@ func (cfg Config) doFetchCloudConfig(proxyAddr string) ([]byte, error) {
 	return ioutil.ReadAll(gzReader)
 }
 
-// updateFrom creates a new Config by merging the given yaml into this Config.
-// Any servers in the updated yaml replace ones in the original Config and any
-// masquerade sets in the updated yaml replace ones in the original Config.
+// updateFrom creates a new Config by 'merging' the given yaml into this Config.
+// The masquerade sets and the collections of servers in the update yaml
+// completely replace the ones in the original Config.
 func (updated *Config) updateFrom(updateBytes []byte) error {
+	// XXX: does this need a mutex, along with everyone that uses the config?
```

Comment:
I believe it's only ever called serially with the updated copy getting passed to all code that uses it, but it's a really good question since clearly things like balancer are using these directly. It's honestly a bit hard to trace all the code paths, but I'll keep looking at it.

We decided not to cut beta5, by the way, as none of us was quite able to get to these code reviews. Thanks for getting them in though. Hopefully it'll go out Monday.